sdk/servicebus/azure-servicebus/azure/servicebus/_common/receiver_mixins.py

```diff
@@ -107,21 +94,24 @@ def _get_source(self):
     def _check_message_alive(self, message, action):
         # pylint: disable=no-member, protected-access
         if message._is_peeked_message:
-            raise MessageSettleFailed(action, ServiceBusMessageError("Messages received by peek can not be settled."))
-        if not self._running:
-            raise MessageSettleFailed(action, ServiceBusMessageError("Orphan message had no open connection."))
+            raise ValueError(
+                "The operation {} is not supported for peeked messages."
+                "Only messages received using receive methods in PeekLock mode can be settled.".format(action)
+            )
+
+        if self._receive_mode == ReceiveMode.ReceiveAndDelete:
+            raise ValueError(
+                "The operation {} is not supported in 'ReceiveAndDelete' receive mode.".format(action)
+            )
+
         if message._settled:
-            raise MessageAlreadySettled(action)
-        try:
-            if message._lock_expired:
```

Comment:
I thought we were keeping some kind of artificial client-side check here until we could fix uamqp?

Reply:
yep, we only check in the message settlement now